aws/resource_aws_eip.go

```diff
@@ -231,13 +231,27 @@ func resourceAwsEipUpdate(d *schema.ResourceData, meta interface{}) error {
 
 	// If we are updating an EIP that is not newly created, and we are attached to
 	// an instance or interface, detach first.
-	if (d.Get("instance").(string) != "" || d.Get("association_id").(string) != "") && !d.IsNewResource() {
+	disassociate := false
+	if !d.IsNewResource() {
+		if (d.Get("instance").(string) != "") && d.HasChange("instance") {
```

Comment:
Nitpicks: The brackets here are unnecessary + I think it would make a bit more sense (when trying to read the code) if it was reversed, i.e.

```go
if d.HasChange("instance") && d.Get("instance").(string) != "" {
```